code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+	list("key1" = "Air injector",				"key2" = "34",	"key3" = "4",	"key4" = "4",	"key5" = "injector",		"key6" = "0"),
+	list("key1" = "Gas pump",					"key2" = "9",	"key3" = "4",	"key4" = "4",	"key5" = "pump",			"key6" = "0"),
+	list("key1" = "Volume pump",				"key2" = "16",	"key3" = "4",	"key4" = "4",	"key5" = "volumepump",		"key6" = "0"),
+	list("key1" = "Passive gate",				"key2" = "15",	"key3" = "4",	"key4" = "4",	"key5" = "passivegate",		"key6" = "0"),
+	list("key1" = "Connector",					"key2" = "4",	"key3" = "4",	"key4" = "4",	"key5" = "connector",		"key6" = "0"),
+	list("key1" = "Gas sensor",					"key2" = "98",	"key3" = "4",	"key4" = "1",	"key5" = "sensor",			"key6" = "0"),
+	list("key1" = "Meter",						"key2" = "99",	"key3" = "4",	"key4" = "1",	"key5" = "meter",			"key6" = "0"),
+	list("key1" = "Gas filter",					"key2" = "13",	"key3" = "4",	"key4" = "4",	"key5" = "filter",			"key6" = "0"),
+	list("key1" = "Gas mixer",					"key2" = "14",	"key3" = "4",	"key4" = "4",	"key5" = "mixer",			"key6" = "0"),
+	list("key1" = "Dual-port vent pump",		"key2" = "36",	"key3" = "4",	"key4" = "2",	"key5" = "dual-port vent",	"key6" = "0"),
+	list("key1" = "Passive vent",				"key2" = "37",	"key3" = "4",	"key4" = "4",	"key5" = "passive vent",	"key6" = "0"),
+	list("key1" = "Straight HE pipe",			"key2" = "2",	"key3" = "5",	"key4" = "2",	"key5" = "he",				"key6" = "0"),
+	list("key1" = "Bent HE pipe",				"key2" = "3",	"key3" = "5",	"key4" = "4",	"key5" = "he",				"key6" = "1"),
+	list("key1" = "Junction",					"key2" = "6",	"key3" = "5",	"key4" = "4",	"key5" = "junction",		"key6" = "0"),
+	list("key1" = "Heat exchanger",				"key2" = "17",	"key3" = "5",	"key4" = "4",	"key5" = "heunary",			"key6" = "0"))
+
+//Pipename, pipe type, number of unique orientations, name of pipe icon
+var/list/dpipelist = list(
+	list("key1" = "Straight pipe",	"key2" = "0",	"key3" = "2",	"key4" = "conpipe-s"),
+	list("key1" = "Bent pipe",		"key2" = "1",	"key3" = "4",	"key4" = "conpipe-c"),
+	list("key1" = "Junction",		"key2" = "2",	"key3" = "4",	"key4" = "conpipe-j1"),
+	list("key1" = "Y-junction",		"key2" = "4",	"key3" = "4",	"key4" = "conpipe-y"),
+	list("key1" = "Trunk",			"key2" = "5",	"key3" = "4",	"key4" = "conpipe-t"),
+	list("key1" = "Bin",			"key2" = "6",	"key3" = "1",	"key4" = "condisposal"),
+	list("key1" = "Outlet",			"key2" = "7",	"key3" = "4",	"key4" = "outlet"),
+	list("key1" = "Chute",			"key2" = "8",	"key3" = "4",	"key4" = "intake"))
+
+//Procs
+
+/obj/item/weapon/rpd/proc/Activaterpd(soundfile, delay)
+	playsound(loc, soundfile, 50, 1)
+	if(prob(15))
+		spark_system.start()
+	if(delay)
+		cooldown = 1
+		spawn(spawndelay)
+		cooldown = 0
+	return
+
+/obj/item/weapon/rpd/proc/Checkifbent(x)
+	if(x in list(1,3,25,27))
+		return 1
+	else
+		return 0
+
+//This proc puts each necessary icon into the initial user's cache for later use. It then sets initialised = 1 to avoid calling itself unnecessarily.
+/obj/item/weapon/rpd/proc/Initialiseicons()
+	var/F
```

Comment:
Why are you defining these variables ahead of time? Define them in the for loop themselves.